public/test/core/redux/reducerTester.ts

```diff
@@ -66,9 +66,9 @@ export const reducerTester = <State>(): Given<State> => {
     disableDeepFreeze = false
   ): When<State> => {
     reducerUnderTest = reducer;
-    initialState = { ...state };
+    initialState = state;
```

Comment:
I think we need to be a bit more clever here. If this is an object, I think we should use cloneDeep from lodash to clone the state.

Reply:
I used the spread syntax before but I realise that using cloneDeep would be better, this means that when I use reducerTester I don't need to worry about spreading the state as a consumer. WDYT?

Reply:
my first take on this was to deep clone the state object (using `JSON.parse(JSON.stringify())` for simplicity, but the same apply to cloneDeep) but it's not clear to me why we want it.

as far as I understood, `initialState` is frozen to "enforce" immutability, so if you don't opt-out from this you should be fine with the freeze leaking over the original object, right?

> this means that when I use reducerTester I don't need to worry about spreading the state as a consumer

Not sure why it would be needed, can you explain please?

Reply:
I'll try to explain this, you could end up in false positive tests if you use the initialState passed to reducerTester in your expects. This is not entirely a freeze issue but a way to prevent false positive tests in certain circumstances. But this is just me being super cautious you need to do pretty bad things to get false positives and as we're using redux toolkit this might never happen. 

```ts
      const dummyAction = createAction<string>('dummyAction');

      const reducer = (state: any, action: AnyAction): any => {
        state.newProp = 'A'; // an example where we add a new prop to state
        return state;
      };

      const initialState = { prop: 'A' };
      reducerTester()
        .givenReducer(reducer, initialState, true, true) // we disable freeze
        .whenActionIsDispatched(dummyAction('some string'))
        .thenStateShouldEqual(initialState); // because initialState is the exact same instance as the reducer state this will not fail but it should as we've added a new prop
```

Reply:
ooh, didn't think about that situation. Thanks!